example/lib/examples/downloader.dart

```diff
@@ -0,0 +1,150 @@
+import 'dart:async';
+
+import 'package:bccm_player/bccm_player.dart';
+import 'package:bccm_player/src/pigeon/downloader_pigeon.g.dart';
+import 'package:bccm_player_example/example_videos.dart';
+import 'package:flutter/material.dart';
+
+class Downloader extends StatefulWidget {
+  const Downloader({super.key});
+
+  @override
+  State<Downloader> createState() => _DownloaderState();
+}
+
+class DownloadState {
+  DownloadState({required this.download, required this.progress});
+
+  Download download;
+  double progress;
+}
+
+class _DownloaderState extends State<Downloader> {
+  late BccmPlayerController controller;
+  List<DownloadState> downloads = [];
+  StreamSubscription<DownloadStatusChangedEvent>? _subscription;
+  bool statusLoopRunning = false;
+
+  void initializeController() async {
+    controller = BccmPlayerController.empty();
+    await controller.initialize();
+    controller.setMixWithOthers(true);
+  }
+
+  void loadDownloads() async {
+    final localDownloads = await DownloaderInterface.instance.getDownloads();
+    final List<DownloadState> result = [];
+    for (var download in localDownloads) {
+      result.add(DownloadState(download: download, progress: await DownloaderInterface.instance.getDownloadStatus(download.key)));
+    }
+    result.sort((a, b) => a.download.key.compareTo(b.download.key));
+
+    setState(() {
+      downloads = result;
+    });
+  }
+
+  void startStatusLoop() async {
+    statusLoopRunning = true;
+
+    while (statusLoopRunning) {
+      await Future.delayed(const Duration(milliseconds: 300));
+
+      final Map<String, double> results = {};
+      for (var state in downloads) {
+        final progress = await DownloaderInterface.instance.getDownloadStatus(state.download.key);
+        results[state.download.key] = progress;
+        debugPrint("P ${state.download.config.title}: ${state.progress}");
+      }
+
+      setState(() {
+        downloads.forEach((state) {
+          state.progress = results[state.download.key]!;
```

Comment:
This status loop is competing with the `downloadStatusEvents` listener during a download since they are both setting the progress. Just a heads up because I got confused (:

Reply:
Yeah, I added this loop because on Android it is not as straightforward, as on iOS, to add an event stream. AVFoundation has a progress callback, while exoplayer does not.

I'm not sure if we really need a stream for the progress. It was not in original api design from the ticket. Maybe we should remove it completely. What do you think?